api/controllers/v1/jobs.ts

```diff
@@ -146,6 +148,22 @@ function limit_message_size(message) {
   return message;
 }
 
+function extractUrlFromMessage(fullDocument) {
+  if (fullDocument.logs.length > 0) {
+    const urls = [];
+    for (let i = 0; i < fullDocument.logs.length; i++) {
+      const element = fullDocument.logs[i];
+      const ret = element.match(/\bhttps?:\/\/\S+/gi);
+      console.log(ret);
+      if (ret) {
+        urls.push(...ret);
+      }
+    }
+    return urls;
+  }
+  return [];
+}
+
```

Comment:
```suggestion
function extractUrlFromMessage(fullDocument) {
  const { logs } = fullDocument;
  let urls = (logs?.length > 0 )
    ? logs.flatMap(log => log.match(/\bhttps?:\/\/\S+/gi) || []);
    : [];
  return urls;
}

```
Optional suggestion for readability + null safety

this seems like a good case for [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map)

repl as PoC: https://replit.com/@casthewiz/BoringVioletTag#index.js